plugin/core/views.py

```diff
@@ -176,20 +176,58 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
     })
 
 
-def minihtml(view: sublime.View, content: Union[str, dict]) -> str:
-    """ Content can be a string or MarkupContent. """
+def minihtml(view: sublime.View, content: Union[str, dict, list], prefer_plain_text: bool) -> str:
+    """
+    Formats provided input into markup accepted by minihtml.
+
+    Content can be in one of those formats:
+     - string: treated as plain text
+     - MarkedString: string | { language: string; value: string }
+     - MarkedString[]
+     - MarkupContent: { kind: MarkupKind, value: string }
+
+    Since we can't distinguish between plain text string and a MarkedString in a string form, we
+    need to be told explicitly which one to use for string input.
+
+    :param view
+    :param content
+    :param prefer_plain_text: Whether to treat *string* input as plain text (True) or markdown (False).
+
+    :returns: Formatted string
+    """
     if isinstance(content, str):
-        return text2html(content)
-    elif isinstance(content, dict):
-        value = content.get("value") or ""
+        return text2html(content) if prefer_plain_text else mdpopups.md2html(view, content)
```

Comment:
I believe you have two places where you return MarkedString `string`
line 199 and line 229

```
        if language:
            # MarkedString (dict)
            return mdpopups.md2html(view, "```{}\n{}\n```\n".format(language, value))
        # MarkedString (string)
        return mdpopups.md2html(view, value)   # this here can be deleted
```

Reply:
True, the second one shouldn't be needed.